downloads/swan-lake-release-notes/swan-lake-2201.3.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,460 @@
+
+A few backward-incompatible changes have been introduced during the Swan Lake Beta program, and thereby, some of your existing packages may not compile with Ballerina 2201.3.0 (Swan Lake). Therefore, you need to delete the `Dependencies.toml` file to force the dependency resolver to use the latest versions of your dependencies. 
+
+## Language updates
+
+### New features
+
+#### Added a new field to the `display` annotation
+
+A new field named `kind` has been introduced to the `display` annotation to indicate the kind of data. Allowed values are `text`, `password`, and `file`.
+
+```ballerina
+public type RefreshTokenGrantConfig record {|
+    @display {
+        iconPath: "Field.icon",
+        label: "clientSecret field",
+        kind: "password"
+    }
+    string clientSecret;
+|};
+```
+
+#### Added support for function pointers with defaultable parameters
+
+Function pointers are now allowed with default values for parameters. Any expression can be used as the default value of a function pointer parameter.
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    int num1 = 100;
+    int num2 = 50;
+    int num3 = 25;
+    function (int a = 0, int b = 0, int c = 0) returns int total = getSum;
+
+    io:println(total()); // Prints `0`.
+    io:println(total(num1, num2)); // Prints `150`.
+    io:println(total(num1, num2, num3)); // Prints `175`.
+}
+
+function getSum(int num1, int num2, int num3) returns int {
+    return num1 + num2 + num3;
+}
+```
+
+### Improvements
+
+#### More improvements on working with optional fields
+
+If there is an optional `T x?;` field in a record, the absence of `x` is represented by nil where `T` does not allow nil.
+
+```ballerina
+import ballerina/io;
+
+type Employee record {
+    int id?;
+    string department?;
+};
+
+public function main() {
+    Employee e = {id: 2, department: "HR"};
+    e.id = ();
+    e = {id: 3};
+    var {id: _, department} = e;
+    io:println(department is ()); // true
+
+    int? idOrNil = ();
+    e = {id: idOrNil, department: "Engineering"};
+    io:println(e.id is ()); // true
+}
+
+```
+
+#### Allow an optional terminating semicolon for module-level declarations
+
+Previously, a closing semicolon was not allowed after the module-level declarations below.
+- block function body
+- service declaration
+- module class definition
+- module enumeration declaration 
+
+Now, optionally, you can end these declarations with a semicolon.
+
+### Backward-incompatible changes
+
+#### Disallow using the `-9223372036854775808` unary expression as an integer
+
+Previously, `-9223372036854775808` was allowed to be assigned to an integer in Ballerina. However, according to the language specification, `-9223372036854775808` is considered a unary expression, which requires the expression after `-` to be a valid value for `int`. Therefore, `-9223372036854775808` now results in a compilation error since `9223372036854775808` is out of range for `int`.
```

Comment:
```suggestion
Previously, `-9223372036854775808` was allowed to be used as an integer in Ballerina. However, according to the language specification, `-9223372036854775808` is considered a unary expression, which requires the expression after `-` to be a valid value for `int`. Therefore, `-9223372036854775808` now results in a compilation error since `9223372036854775808` is out of range for `int`.
```